runtime/onert/core/src/compiler/train/TensorRegistries.h

```diff
@@ -93,6 +95,15 @@ class TensorRegistries
     return nullptr;
   }
 
+  void iterateTrainableTensors(
+    const std::function<void(const ir::OperandIndex &, backend::train::ITrainableTensor *)> &fn)
+    const
+  {
+    assert(_tensor_regs.size() == 2); // training backend and built-in backcend
```

Comment:
@ragmani Is it possible to have other backend except for training backend and built-in backend? Please let me know.

I found typo in comment. I will fix typo first before your reply.

Reply:
I didn't talk about possibility to have other backend except for training backend and built-in backend. My question was why "this function" checks whether the number of backends is 2. I think there seem to be no problem with iterating trainable tensors even if another backend is introduced.

Reply:
I agree to @ragmani. I don't think so that we should limit the backend count and type. Why don't we leave the possibility of extensibility?

Reply:
No, I don't limit the possibility of extensibility. It just make sure the supported backend at this moment. It is like "Not supported yet". What happens if someone tries to use 3 backends at this moment?

Reply:
I mean that this assertion is suitable for this function because newly introduced backend would have `ITrainableTensor`. This function just iterates `ITranableTensor`s.
Someone may be confused as to whether the number of backends should be 2 **in this function** if someone try to use 3 backends.

Reply:
@ragmani You mean that the location of `assert` is not suitable. Okay. I guess the check is already done in early stage. I will remove this assert.

Reply:
If you want to be defensive, how about finding suitable location and adding this assertion there?

Reply:
I already removed it. : )